wallet/src/Pos/Wallet/Web/Server/Handlers.hs

```diff
@@ -30,27 +31,28 @@ import           Pos.Wallet.Web.Tracking (fixingCachedAccModifier)
 
 servantHandlersWithSwagger
     :: MonadFullWalletWebMode ctx m
-    => (forall x. m x -> Handler x)
+    => (TxAux -> m Bool)
+    -> (forall x. m x -> Handler x)
     -> Server A.WalletSwaggerApi
-servantHandlersWithSwagger nat =
-    hoistServer A.walletApi nat servantHandlers
+servantHandlersWithSwagger submitTx nat =
+    hoistServer A.walletApi nat (servantHandlers submitTx)
    :<|>
     swaggerSchemaUIServer swaggerSpecForWalletApi
 
 ----------------------------------------------------------------------------
 -- The wallet API
 ----------------------------------------------------------------------------
 
-servantHandlers :: MonadFullWalletWebMode ctx m => ServerT A.WalletApi m
-servantHandlers = toServant' A.WalletApiRecord
+servantHandlers :: MonadFullWalletWebMode ctx m => (TxAux -> m Bool) -> ServerT A.WalletApi m
+servantHandlers submitTx = toServant' A.WalletApiRecord
     { _test        = testHandlers
     , _wallets     = walletsHandlers
     , _accounts    = accountsHandlers
     , _addresses   = addressesHandlers
     , _profile     = profileHandlers
-    , _txs         = txsHandlers
+    , _txs         = txsHandlers submitTx
     , _update      = updateHandlers
-    , _redemptions = redemptionsHandlers
+    , _redemptions = redemptionsHandlers submitTx
```

Comment:
Here's a real example of how not-error-prone argument passing is. These handlers (`txsHandlers` and `redemptionsHandlers`) need a `TxAux -> m Bool`, and voila, there is precisely one value of that type in scope.